### ISSUE TITLE
Number formats in a treemap chart? 

### ISSUE THREAD

**Issue by Calandal95** [amcharts/amcharts4#118]
Hi all,
i've got a question about treemap charts:
How can i set number formats differently for name and values in a treemap chart?
Moreover the graph doesn't recognize localization according to the imported locale (i use it_IT)

```
import locale_it_IT from "./it_IT";
...
chart.language.locale = locale_it_IT;
```
Thank you in advance
Regards Alessio

**Comment by martynasma** [comment]
Hi Alessio,

I'm not entirely sure what you mean number format for name? Maybe you can share an an example?

As a side note, you can apply number formats for each placeholder individually. Refer to this section in docs:

https://www.amcharts.com/docs/v4/concepts/formatters/formatting-numbers/#Overriding_format

As for the Italian locale, is it possible to see your file? We don't have it bundled officially, so I'm assuming it's something you have made.

**Comment by Calandal95** [comment]
In my treemap chart i have two datafields, one for the name (but it contains a date) and one for the value. At the top i have this instruction     
 
```
chart.language.locale = locale_it_IT;
chart.numberFormatter.formatNumber = '#,###.##'
...
chart.dataFields.name = 'YEAR';    //YEAR contains: [2016, 2017, 2018]
chart.dataFields.value = 'TOTAL';  // TOTAL contains [100200, 123456, 245960]
...
chart.dataFields.numberFormatter = new am4core.NumberFormatter();
chart.dataFields.numberFormat = "#";
```

The result in the tree map chart shows: 
**YEAR formatted as 2,016 , 2,017, 2,018
TOTAL as 100,200, 123,456, 245,960**

Our italian locale file is saved as it_IT, correctly imported in the component and uses the following separators:

```
    "_decimalSeparator": ",",
    "_thousandSeparator": ".",
```

For other chart types number formatting for both labels and values correctly works as expected.

Thanks in advance
Alessio

**Comment by martynasma** [comment]
Is this just a typo?

```
chart.dataFields.numberFormatter = new am4core.NumberFormatter();
```

`dataFields` does not have `numberFormatter` property, so, if it's not a typo but actually what you have, it is ignored.

I suppose in this case, you might need to use format overriding feature, I mentioned:

```
chart.series.template.columns.template.tooltipText = "{parentName.formatNumber('#')} {name}: {value}";
```

**Comment by martynasma** [comment]
Number formatting is automatically applied only to those data properties that the chart knows to be numbers, like "value". "Name" is not supposed to be number, so the chart does not try to apply the format, which you can force anyway using `formatNumber()` function.

I hope that makes sense :)

> And regarding the format overriding, this will work only for tooltipText, right? I have to format also the bullet?

`tooltipText` is only for tooltip. But this can be similarly used for labels as well:

```
...text = "{value.formatNumber('#')}";
```

**Comment by Calandal95** [comment]
Hello, 

> Number formatting is automatically applied only to those data properties that the chart knows to be numbers, like "value". 

But this will not explain why the chart formatting is ignored and the value field is formatted like this: 

> TOTAL as **100,200, 123,456, 245,960**

Thanks!
Alessio

**Comment by martynasma** [comment]
OK, sorry for not catching this earlier. It should be `numberFormatter.numberFormat` (not `formatNumber`)

I know it might be confusing since in override functions the method is `formatNumber()`.

**Comment by Calandal95** [comment]
I've applied the correction but the formatting in the treemap chart (neihter for the number values only) is still not working.

```
import locale_it_IT from "./it_IT";
chart.language.locale = locale_it_IT;
chart.numberFormatter.numberFormat = '#,###.##';
```
Those are the same instructions used in columns and pie charts. So i can't really understand why it's not working.

Thanks!
Alessio

**Comment by martynasma** [comment]
@Calandal95 

I'm terribly sorry for not getting back about this earlier.

Indeed, for some reason number formatting does not work on values on a Treemap chart.

We're investigating this now.

Will keep you posted.

**Comment by martynasma** [comment]
Hm, I seem to be spoken too soon. It seems that formatting IS being applied correctly to values on Treemap.

Here's an example:
https://codepen.io/team/amcharts/pen/04e064227bfb26c1a9e526c94e75c1e6/

I intentionally set `numberFormat` to some weird one to make it prominent it works.

Let me know if your setup is different.

Or do you mean that separator settings that are in Italian translation file are not working? In that case, can you please share your translation file?

**Comment by martynasma** [comment]
Can you share the complete `it_IT` file?

**Comment by Calandal95** [comment]
Sure! Here it is.
[it_IT.zip](https://github.com/amcharts/amcharts4/files/2236320/it_IT.zip)

**Comment by Calandal95** [comment]
Sure, here it is.
[it_IT.txt](https://github.com/amcharts/amcharts4/files/2236352/it_IT.txt) 
(The file has been renamed to txt just for the purpose of attach it here).

Alessio

**Comment by martynasma** [comment]
Thanks Alessio.

It seems there's a bug in TreeMapSeries that does not pass on correct locale to the number formatter.

Will let you know when the fix is available.

**Comment by martynasma** [comment]
This has now been fixed in dev and will be available in the next official release later this week.

Thank you for your patience!

**Comment by Calandal95** [comment]
> This has now been fixed in dev and will be available in the next official release later this week.
 Thank you for your patience!
 
Thank you for your good work!
Alessio

**Comment by martynasma** [comment]
Fixed in [4.0.0-beta.40](https://github.com/amcharts/amcharts4/releases/tag/4.0.0-beta.40).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by parth22** [comment]
in Flutter i want currency format for Italian currency for that i write one number format functionality but it doesn't allow me to add "," as fractional part in value.

class CurrencyInputFormatter extends TextInputFormatter{
  @override
  TextEditingValue formatEditUpdate(
      TextEditingValue oldValue, TextEditingValue newValue) {
    // TODO: implement formatEditUpdate
    if (newValue.selection.baseOffset == 0) {
      print(true);
      return newValue;
    }
    double value = double.parse(newValue.text);
    final formatter = new NumberFormat("#,###.##", "it_IT");
    String newText = formatter.format(value);
    return newValue.copyWith(
        text: newText,
        selection: new TextSelection.collapsed(offset: newText.length));
  }
}
